app-libs/stf/src/best_energy_helpers.rs

```diff
@@ -9,25 +9,24 @@ use std::{format, fs, vec::Vec};
 pub static ORDERS_DIR: &str = "./records/orders";
 pub static RESULTS_DIR: &str = "./records/market_results";
 
-pub fn get_orders_index(
+pub fn get_merkle_proof_for_actor_from_file(
 	timestamp: &str,
 	actor_id: &str,
-) -> Result<(Vec<Vec<u8>>, usize), StfError> {
+) -> Result<MerkleProofWithCodec<H256, Vec<u8>>, StfError> {
+	let orders = read_orders(timestamp)?;
+
+	get_merkle_proof_for_actor(actor_id, &orders)
+		.ok_or(StfError::Dispatch(format!("Leaf Index error: {:?}", actor_id)))
+}
```

Comment:
I was a bit confused. I prepared everything for you, was there a reason that you did not use it?

Reply:
I changed this function to directly return a proof, so you don't need to care about it in the `getter.rs`.